flow_runner.go

```diff
@@ -14,79 +13,83 @@ import (
 
 type flowRunner struct {
 	output      io.Writer
-	params      Params
+	params      map[string]parameter
+	paramValues map[string]Value
 	tasks       map[string]Task
-	verbose     bool
+	verbose     *BoolParam
 	defaultTask RegisteredTask
 }
 
 // Run runs provided tasks and all their dependencies.
 // Each task is executed at most once.
 func (f *flowRunner) Run(ctx context.Context, args []string) int {
-	// prepare flag.FlagSet
-	cli := flag.NewFlagSet("", flag.ContinueOnError)
-	cli.SetOutput(f.output)
-	verbose := cli.Bool("v", false, "Verbose output: log all tasks as they are run. Also print all text from Log and Logf calls even if the task succeeds.")
-	usage := func() {
-		fmt.Fprintf(cli.Output(), "Usage: [flag(s)] [key=val] task(s)\n")
-		fmt.Fprintf(cli.Output(), "Flags:\n")
-		cli.PrintDefaults()
-
-		fmt.Fprintf(cli.Output(), "Tasks:\n")
-		w := tabwriter.NewWriter(cli.Output(), 1, 1, 4, ' ', 0)
-		keys := make([]string, 0, len(f.tasks))
-		for k, task := range f.tasks {
-			if task.Description == "" {
-				continue
-			}
-			keys = append(keys, k)
-		}
-		sort.Strings(keys)
-		for _, k := range keys {
-			t := f.tasks[k]
-			fmt.Fprintf(w, "  %s\t%s\n", t.Name, t.Description)
-		}
-		w.Flush() //nolint // not checking errors when writting to output
-
-		if f.defaultTask.name != "" {
-			fmt.Fprintf(cli.Output(), "Default task: %s\n", f.defaultTask.name)
-		}
+	if unusedParams := f.unusedParams(); len(unusedParams) > 0 {
+		fmt.Fprintf(f.output, "unused parameters: %v\n", unusedParams)
+		return CodeUnusedParams
```

Comment:
I think we should `panic` here instead.
if it happens it is clearly a coding error and the `Taskflow` would never return anything else.
It would also follow the same pattern as it is used in `Register*Param`.

```suggestion
		panic(fmt.Sprintf("unused parameters: %v", unusedParams)
```

TBH in the next PR, I want to remove the current `Register` method and then rename the `MustRegister` to `Register`.

Reply:
Regarding Register/MustRegister: Is there a need to support a RegisterTask/Param with error response? I'd rather guess if unused params are a panic, why not also for duplicate names?
Is there a need to be able to continue the program if registration errors occur?
Otherwise I'd opt to minimize the API surface.

Reply:
Although, re-reading your comment I believe this is also what you wanted: Have all Register() methods panic - no error cases.

Reply:
return code removed again, using panic instead

Reply:
> Although, re-reading your comment I believe this is also what you wanted: 

Correct :wink: